lib/3scale_toolbox/proxy_logger.rb

```diff
@@ -0,0 +1,21 @@
+module ThreeScaleToolbox
+  class ProxyLogger
+    instance_methods.each do |meth|
+      # skipping undef of methods that "may cause serious problems"
+      undef_method(meth) if meth !~ /^(__|object_id)/
+    end
+
+    def initialize(subject)
+      @subject = subject
+    end
+
+    def method_missing(name, *args)
+      start_time = Time.now
```

Comment:
Better to use Benchmark module for this.

Reply:
you mean something like this?? 
```ruby
def method_missing(name, *args)                                                                   
  result = nil                                                                                    
  time = Benchmark.realtime do                                                                    
    result = @subject.send(name, *args)                                                           
  end        
ensure                                                                                     
  puts "-- call #{name} args |#{args.inspect[0..80]}| response |#{result.inspect[0..80]}| - (#{time}s)"
end                                                                                               
```

I do not see any enhancement doing that. Besides, if error raised from subject this implementation does not print time elapsed, while current one does.

Reply:
The first enhancement is that Benchmark.realtime is measuring the time properly, using monotonic time.

If you care about the error case, then just use monotonic time.